src/org/traccar/Config.java

```diff
@@ -15,48 +15,55 @@
  */
 package org.traccar;
 
+import org.apache.commons.lang3.StringUtils;
+
 import java.io.FileInputStream;
 import java.io.IOException;
 import java.io.InputStream;
+import java.util.Enumeration;
 import java.util.Properties;
 
 public class Config {
 
     private final Properties properties = new Properties();
-    private Properties defaultProperties;
 
-    public void load(String file) throws IOException {
+    void load(String file) throws IOException {
         try (InputStream inputStream = new FileInputStream(file)) {
             properties.loadFromXML(inputStream);
         }
 
         String defaultConfigFile = properties.getProperty("config.default");
         if (defaultConfigFile != null) {
             try (InputStream inputStream = new FileInputStream(defaultConfigFile)) {
-                defaultProperties = new Properties();
+                Properties defaultProperties = new Properties();
                 defaultProperties.loadFromXML(inputStream);
+
+                Enumeration props = defaultProperties.propertyNames();
+                while (props.hasMoreElements()) {
```

Comment:
Actually that part is not related to environment variables.
Instead of holding properties into two separate variables `properties` and `defaultProperties` the idea was to merge everything and use only `properties`. That way